### PR TITLE
Add initial resource view

### PR DESCRIPTION
**Pull request by anthonydresser**
Adds the initial resource view. Resources are declares in the contributions of extensions that are then rendered in a flat tree. Detects the arc extension and only registers the view if it is installed.

Current:
Untitled

TODO:
Add icons

### CODE REVIEW

File: extensions/azurecore/package.json [hunk 1]
```diff
@@ -19,6 +19,13 @@
     "url": "https://github.com/Microsoft/azuredatastudio.git"
   },
   "contributes": {
+    "resourceViewResources": [
```
Comment (kburtram) [inline extensions/azurecore/package.json:22]: did we want this to only show when the user has the azurearc extension installed in the Sep release?  In that case, should we add this contribution to the azure arc extension?

File: src/sql/workbench/contrib/resourceViewer/browser/media/resourceViewerView.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+.resource-view {
```
Comment (kburtram) [inline src/sql/workbench/contrib/resourceViewer/browser/media/resourceViewerView.css:6]: this name is a little generic (i.e. I could imagine someone adding a conflicting ID in the future).  Maybe something like `resource-explorer-view` would be better scoped?

File: src/sql/workbench/contrib/resourceViewer/browser/resourceViewer.contribution.ts [hunk 3]
```diff
@@ -36,3 +45,21 @@ const resourceViewerDescriptor = EditorDescriptor.create(
...
+
+function registerResourceViewerContainer() {
+	const viewContainer = Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry).registerViewContainer({
+		id: RESOURCE_VIEWER_VIEW_CONTAINER_ID,
+		name: localize('resourceViewer', "Resource Viewer"),
+		ctorDescriptor: new SyncDescriptor(ResourceViewerViewlet),
+		icon: Codicon.database.classNames,
+		alwaysUseContainerInfo: true
+	}, ViewContainerLocation.Sidebar);
+	// registry.registerWorkbenchAction(SyncActionDescriptor.from(OpenDebugViewletAction, { primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KEY_D }), 'View: Show Run and Debug', nls.localize('view', "View"));
```
Comment (kburtram) [inline src/sql/workbench/contrib/resourceViewer/browser/resourceViewer.contribution.ts:60]: dead code

File: src/sql/workbench/contrib/resourceViewer/browser/resourceViewerView.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+		this.listContainer = append(container, $('.resource-view'));
+
+		const renderers: ITreeRenderer<TreeElement, any, any>[] = [new ResourceRenderer()];
+
+		this.model = new TreeModel();
+
+		this.tree = this.instantiationService.createInstance(
+			WorkbenchDataTree,
+			'Resource View',
```
Comment (kburtram) [inline src/sql/workbench/contrib/resourceViewer/browser/resourceViewerView.ts:60]: should this be localizable?

**Review by kburtram — APPROVED** [review]
LGTM, though I think the viewer is planned to be enabled through Arc for Sep.

File: src/sql/workbench/contrib/resourceViewer/browser/resourceViewerView.ts (hunk 4 above)
Comment (anthonydresser) [reply src/sql/workbench/contrib/resourceViewer/browser/resourceViewerView.ts:60]: No this is an internal identifier

File: extensions/azurecore/package.json (hunk 1 above)
Comment (anthonydresser) [reply extensions/azurecore/package.json:22]: This is how resources get contributed. If we want to view to only show when an extension is installed I can do that separately.

**Pull request merged**